src/libraries/Microsoft.PowerFx.Interpreter/Environment/SymbolValues.cs

```diff
@@ -12,19 +13,20 @@ namespace Microsoft.PowerFx
     /// Mutable collection for runtime Values for a <see cref="SymbolTable"/>.
     /// Will lazily create a symbol table. 
     /// To match to an existing symbol table, call <see cref="SymbolExtensions.CreateValues(ReadOnlySymbolTable, ReadOnlySymbolValues[])"/>.
-    /// </summary>
+    /// </summary>
+    [NotThreadSafe]
     public class SymbolValues : ReadOnlySymbolValues
     {
         // Map of name Slots --> Values. 
         // Index by Slot.SlotIndex. This could be optimized to be a dense array. 
-        private readonly Dictionary<int, Tuple<ISymbolSlot, FormulaValue>> _symbolValues = new Dictionary<int, Tuple<ISymbolSlot, FormulaValue>>();
+        private readonly ConcurrentDictionary<int, Tuple<ISymbolSlot, FormulaValue>> _symbolValues = new ConcurrentDictionary<int, Tuple<ISymbolSlot, FormulaValue>>();
```

Comment:
>ConcurrentDictionary [](http://example.com/codeflow?start=25&length=20)

We shouldn't preemptively change to ConcurrentDictionary  - if it's not thread safe, don't pretend that it is.  

Instead, add guards on Add(), etc to help ensure.